src/main/java/com/exasol/releasedroid/github/GitHubPlatformValidator.java

```diff
@@ -17,33 +17,36 @@
     protected static final String GITHUB_WORKFLOW_PATH = ".github/workflows/github_release.yml";
     private static final Logger LOGGER = Logger.getLogger(GitHubPlatformValidator.class.getName());
     private final GithubGateway githubGateway;
+    private final Repository repository;
 
     /**
      * Create a new instance of {@link GitHubPlatformValidator}.
      *
+     * @param repository    repository to validate
      * @param githubGateway instance of {@link GithubGateway}
      */
-    public GitHubPlatformValidator(final GithubGateway githubGateway) {
+    public GitHubPlatformValidator(final Repository repository, final GithubGateway githubGateway) {
+        this.repository = repository;
         this.githubGateway = githubGateway;
     }
 
     @Override
     // [impl->dsn~validate-github-workflow-exists~1]
-    public Report validate(final Repository repository) {
+    public Report validate() {
         LOGGER.fine("Validating GitHub-specific requirements.");
         final Report report = Report.validationReport();
-        final String version = repository.getVersion();
-        final ReleaseLetter releaseLetter = repository.getReleaseLetter(version);
-        report.merge(validateChangesFile(repository, releaseLetter));
-        report.merge(validateFileExists(repository, GITHUB_WORKFLOW_PATH, "Workflow for a GitHub release."));
+        final String version = this.repository.getVersion();
+        final ReleaseLetter releaseLetter = this.repository.getReleaseLetter(version);
+        report.merge(validateChangesFile(releaseLetter));
+        report.merge(validateFileExists(this.repository, GITHUB_WORKFLOW_PATH, "Workflow for a GitHub release."));
```

Comment:
I would suggest to remove this parameter, as we now have access to the repo.
```suggestion
        report.merge(validateFileExists(GITHUB_WORKFLOW_PATH, "Workflow for a GitHub release."));
```

Reply:
validateFileExists() method is in the abstract class. In the implementation, we have different types of repositories: Java, Scala. So I also can't move the Repository field to the abstract class.